src/components/CommentList.jsx

```diff
@@ -1,14 +1,32 @@
+import { useEffect, useState } from "react";
 import CommentCard from "./CommentCard";
+import { getComments } from "../app";
 
-const CommentList = ({ comments }) => {
-  return (
-    <section id="comment-list">
-      <h4>Comments:</h4>
-      {comments.map((comment) => {
-        return <CommentCard key={comment.comment_id} comment={comment} />;
-      })}
-    </section>
-  );
+const CommentList = ({ article_id }) => {
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  useEffect(() => {
+    setIsLoading(true);
+    getComments(article_id).then((fetchedComments) => {
+      setComments(fetchedComments);
+      setIsLoading(false);
+    });
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading comments...</p>;
+  } else if (!comments.length) {
+    return <p>No comments yet</p>;
+  } else {
```

Comment:
These statements have returns in them so you dont actually need to specifically have an else if or an else at the very end as that code is not reachable anyway.